src/promptflow-evals/tests/evals/e2etests/test_builtin_evaluators.py

```diff
@@ -11,8 +11,8 @@
 
 
 @pytest.mark.usefixtures("recording_injection", "vcr_recording")
-@pytest.mark.e2etest
```

Comment:
Can we apply @pytest.mark.localtest at the class level and override only needed test with @pytest.mark.azuretest? If this way works, we don't have to specify mark for each test.

Reply:
The marks in the python are not mutually exclusive. Consider example:
```
import pytest

@pytest.mark.foo
class TestPytest:

    @pytest.mark.bar
    def test_logger(self, caplog):
        assert False

    def test_logger2(self, caplog):
        assert True
```
In this case calling `pytest -m foo test_mark.py` will result in failure:
![image](https://github.com/user-attachments/assets/12b254ac-dc0b-4551-81ca-365b727e6794)

Reply:
Thanks for verifying! Maybe we can use a pytest hook to filter the markers in `conftest.py`. If both markers are present, remove the localtest.

def pytest_collection_modifyitems(items):
    for item in items:
        if item.get_closest_marker('azuretest'):
            item.own_markers = [marker for marker in item.own_markers if marker.name != 'localtest']

Reply:
Fixed.